Lib/test/test_concurrent_futures/executor.py

```diff
@@ -73,52 +73,48 @@ def test_map_timeout(self):
 
         self.assertEqual([None, None], results)
 
-    def test_map_buffersize(self):
-        integers = range(8)
-
+    def test_map_buffersize_validation(self):
         with self.assertRaisesRegex(
             ValueError,
             "buffersize must be None or > 0",
-            msg="`Executor.map` should raise if `buffersize` is not positive.",
         ):
-            self.executor.map(str, integers, buffersize=0)
-
-        for buffersize in (1, 2, len(integers), len(integers) * 2):
-            self.assertEqual(
-                list(self.executor.map(str, integers, buffersize=buffersize)),
-                list(map(str, integers)),
-                msg="`Executor.map` with `buffersize` should behave the same as `map`.",
-            )
-
-            self.assertEqual(
-                list(self.executor.map(add, integers, integers, buffersize=buffersize)),
-                list(map(sum, zip(integers, integers))),
-                msg="`Executor.map` with `buffersize` should work correctly on multiple input iterables",
-            )
-
-            self.assertEqual(
-                next(self.executor.map(str, itertools.count(), buffersize=buffersize)),
-                next(map(str, integers)),
-                msg="`Executor.map` with `buffersize` should work correctly on an infinite input iterator.",
-            )
-
-            self.assertFalse(
-                list(self.executor.map(str, [], buffersize=buffersize)),
-                msg="`Executor.map` with `buffersize` should return an empty iterator if the input iterable is empty.",
-            )
-
-            self.assertFalse(
-                list(self.executor.map(str, buffersize=buffersize)),
-                msg="`Executor.map` with `buffersize` should return an empty iterator if no input iterable is provided.",
-            )
-
-        integers_iter = iter(integers)
-        self.executor.map(str, integers_iter, buffersize=buffersize)
-        self.executor.shutdown(wait=True)  # wait for pending tasks to complete
+            self.executor.map(str, range(4), buffersize=0)
```

Comment:
- You can also test for a negative buffer size
- Maybe test for a non-integer buffersize? (`timeout` implicitly raises a TypeError if we give a non-numeric value when we do `timeout + monotonic()` but we should also check that passing buffersize="abc" would raise a TypeError as well, either implicitly or explicitly)

Reply:
Makes total sense!
- added the ValueError test for negative buffersize

Regarding the type:
-  when giving it a string: we get an implicit `TypeError: '<' not supported between instances of 'str' and 'int'` via the check `buffersize < 1`
- when giving it a float: we get a `ValueError: Stop argument for islice() must be None or an integer: 0 <= x <= sys.maxsize` `islice(zipped_iterables, buffersize)`, so not a TypeError. What is the guideline here? I'm confused because strictly speaking `islice` should itself return a `TypeError`, right?

Reply:
Should we add with a 
```python
if not isinstance(bufsize, int):
    raise TypeError("buffersize must be an integer")
```